tests/main/dbus-activation-session-legacy/task.yaml

```diff
@@ -0,0 +1,39 @@
+summary: D-Bus session services support activation without a user of instance systemd
+
+systems: [-ubuntu-core-*]
+
+prepare: |
+    snap set system experimental.user-daemons=true
+    snap set system experimental.dbus-activation=true
+
+restore: |
+    if [ -f dbus-launch.pid ]; then
+        kill "$(cat dbus-launch.pid)"
+        rm -f dbus-launch.pid
+    fi
+    snap unset system experimental.dbus-activation
+    snap unset system experimental.user-daemons
+
+execute: |
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB"/snaps.sh
+
+    echo "Set up a private D-Bus session bus"
+    eval "$(dbus-launch --sh-syntax)"
```

Comment:
I'm not super fond of seeing this code. Could we run it in a systemd service so that, at least, we can reliably shut it down, along with all children?

Reply:
Having gone through the exercise I would like you to do something like this instead: https://github.com/zyga/snapd/blob/tweak/detect-stray-dbus-daemon/tests/main/invariant-tool/task.yaml#L64